tests/headers/c-empty-layout.h

```diff
@@ -0,0 +1 @@
+struct Foo {};
```

Comment:
I know we have tests for this intermingled here and there elsewhere, but it would be nice to have a C++ companion for this test that has the same contents but with a `.hpp` extension.